.gitattributes

```diff
@@ -1,4 +1,5 @@
 * text=auto
 
 # Test files should not have their line endings modified by git
-/tests/languages/**/*.test binary
\ No newline at end of file
+/tests/languages/**/*.test binary
+/tests/languages/**/*.testhtml binary
```

Comment:
Line ends. Git will automatically checks out and commits line ends in whatever format is configured on your system. That's a problem for test files because the output now depends on your git config. We also want to be able to have tests that test code with different line ends.

Reply:
> We also want to be able to have tests that test code with different line ends.

I'd suggest that's a poor reason to make ALL the tests `binary` though.  If the highlighter is expected to pass text thru with line endings preserved then surely several unit tests [coded in pure JS perhaps] (of various line encodings) could confirm this behavior and the behavior for the majority of the test suite could be to expect a common line encoding or to normalize the line encodings.

In working on a few projects where test files were purposely supposed to have weird encodings (or hidden space at the end of lines - when editors default to removing this) it's always been a huge pain.  Line endings and whitespace simply shouldn't break things.

Reply:
Frankly, you're right. We can just have the test file parser auto-convert line ends and that's it. We sorta require `\r\n` line ends for these files anyway.

That will be a fun PR. Changing the line ends in hunderts of files... Well, I'll do that tomorrow.

Reply:
We recently went thru this sort of thing with [Wren](https://wren.io) as well (my new obsession).  The compiler/parser would treat different line endings very differently.  You'd mostly notice this inside multi-line strings where the value of the string could vary depending on how the file would saved... this was very counter intuitive - and we fixed it.

In my testing Ruby, Node.js, and Python all of them "normalize" the script input to a standardized line encoding (regardless of the on-disk encoding used)... so that whatever line encoding the script runs in the behavior is always consistent.  

> Changing the line ends in hunderts of files

Good luck!